### PR TITLE
#239: Only allow audio files for song uploading

### PR DESCRIPTION
**Pull request by perig99**
## Description
 - Updated the file upload logic to validate audio files more accurately.
 - Updated unit test for the AddSongPlayListAccordion component to validate file input handling.

## Commit type
  - `feat`: Only allow audio files for song uploading

## Issue
Issue #239 

## Solution
This pull request resolves the problem by improving the validation and handling of file uploads, specifically for audio files:
- by changing <input accept="audio/mp3"> attribute to "audio/*"
- by handling unsupported files in handleChangeFile

### CODE REVIEW

File: Electron/src/__tests__/components/Sidebar/AddSongPlayListAccordion.test.tsx [hunk 1]
```diff
@@ -188,10 +188,15 @@ test('AddSongPlaylistAccordion submit song correct', async () => {
...
+  const fileInputElement = component.getByTestId(
+    'sidebar-file-input',
+  ) as HTMLInputElement;
 
-  // Create a sample file
-  const file = new File(['(⌐□_□)'], 'sample.mp3', { type: 'audio/mp3' });
+  // Create a sample files
+  const mp3File = new File(['(⌐□_□)'], 'sample.mp3', { type: 'audio/mp3' });
+  const wavFile = new File(['(⌐□_□)'], 'sample.wav', { type: 'audio/wav' });
+  const flacFile = new File(['(⌐□_□)'], 'sample.wav', { type: 'audio/wav' });
```
Comment (AntonioMrtz) [inline Electron/src/__tests__/components/Sidebar/AddSongPlayListAccordion.test.tsx:198]: Maybe this is a typo, shouldn't it be audio/flac?

File: Electron/src/__tests__/components/Sidebar/AddSongPlayListAccordion.test.tsx [hunk 2]
```diff
@@ -207,9 +212,30 @@ test('AddSongPlaylistAccordion submit song correct', async () => {
       target: { value: 'Rock' },
     });
     fireEvent.change(dropdown, { target: { value: 'Rock' } });
-    fireEvent.change(fileInputElement, { target: { files: [file] } });
   });
 
+  // Valid files
```
Comment (AntonioMrtz) [inline Electron/src/__tests__/components/Sidebar/AddSongPlayListAccordion.test.tsx:217]: Should we have different tests for each case (mp3,flac,wav) ? Example: test_flac, test_invalid_file etc?

File: Electron/src/__tests__/components/Sidebar/AddSongPlayListAccordion.test.tsx
Comment (AntonioMrtz) [inline Electron/src/__tests__/components/Sidebar/AddSongPlayListAccordion.test.tsx:1]: Can we have support for `.aac` files? I feel it's also one of the most used formats

**Review by AntonioMrtz — CHANGES_REQUESTED** [review]
The code is great :) . Thanks for sharing your time and following our conventions and workflows. I did write a few minor improvements that can be done. Let me know if you need anything. Backend pipeline is failing because of an enviroment value only provided in the main repo so don't worry about it.

File: Electron/src/__tests__/components/Sidebar/AddSongPlayListAccordion.test.tsx (hunk 1 above)
Comment (perig99) [reply Electron/src/__tests__/components/Sidebar/AddSongPlayListAccordion.test.tsx:198]: Sorry my bad. Yes it is a typo. I will fix it.

File: Electron/src/__tests__/components/Sidebar/AddSongPlayListAccordion.test.tsx
Comment (perig99) [reply Electron/src/__tests__/components/Sidebar/AddSongPlayListAccordion.test.tsx:1]: I will add support for .aac files.

File: Electron/src/__tests__/components/Sidebar/AddSongPlayListAccordion.test.tsx (hunk 2 above)
Comment (perig99) [reply Electron/src/__tests__/components/Sidebar/AddSongPlayListAccordion.test.tsx:217]: Sorry, but I don't understand this.

**Review by AntonioMrtz — APPROVED** [review]
LGTM :)

**Comment by AntonioMrtz** [comment]
Hi @perig99 , code looks good 👍. Thanks for sharing your time and efforts to contribute to the project. I hope too see you soon in the project, are you looking for other issues to contribute? After I merge I will add you in README and CONTRIBUTORS :)

**Comment by perig99** [comment]
Hi @AntonioMrtz,
thanks for opportunity to take a part on this project. It was my first time experience to contribute on project like this and it was great experience for me. I am focusing on my Master thesis right now. So once, when I will have more free time, I would like to help. :)

**Comment by AntonioMrtz** [comment]
Hi @perig99 I wish you the best on your master dregree! . Feel free to drop by the project whenever you like. We'll be waiting :)

**Pull request merged**